File: content/src/main/java/org/sandboxpowered/silica/minecraft/blocks/FireBlock.java
package org.sandboxpowered.silica.minecraft.blocks;

import org.sandboxpowered.api.block.BaseBlock;
import org.sandboxpowered.api.block.Block;
import org.sandboxpowered.api.state.BlockState;
import org.sandboxpowered.api.state.Properties;
import org.sandboxpowered.api.state.StateFactory;

public class FireBlock extends BaseBlock {
    public FireBlock(Settings settings) {
        super(settings);
    }

    @Override
    public void appendProperties(StateFactory.Builder<Block, BlockState> builder) {
        super.appendProperties(builder);
        builder.add(Properties.AGE_15, Properties.EAST, Properties.NORTH, Properties.SOUTH, Properties.WEST, Properties.UP);
    }
}

File: common/src/main/kotlin/org/sandboxpowered/silica/block/FireBlock.kt
package org.sandboxpowered.silica.block

import org.sandboxpowered.silica.block.SilicaBlockProperties.AGE_15
import org.sandboxpowered.silica.block.SilicaBlockProperties.EAST
import org.sandboxpowered.silica.block.SilicaBlockProperties.NORTH
import org.sandboxpowered.silica.block.SilicaBlockProperties.SOUTH
import org.sandboxpowered.silica.block.SilicaBlockProperties.UP
import org.sandboxpowered.silica.block.SilicaBlockProperties.WEST
import org.sandboxpowered.silica.state.StateProvider
import org.sandboxpowered.silica.state.block.BlockState
import org.sandboxpowered.silica.util.Identifier

class FireBlock(identifier: Identifier) : BaseBlock(identifier) {
    override fun appendProperties(builder: StateProvider.Builder<Block, BlockState>) {
        super.appendProperties(builder)
        builder.add(AGE_15, UP, NORTH, EAST, SOUTH, WEST)
    }
}